patches/net/minecraft/world/item/component/ItemContainerContents.java.patch

```diff
@@ -1,9 +1,25 @@
 --- a/net/minecraft/world/item/component/ItemContainerContents.java
 +++ b/net/minecraft/world/item/component/ItemContainerContents.java
-@@ -164,6 +_,35 @@
+@@ -54,6 +_,12 @@
          }
      }
  
++    public ItemContainerContents with(int index, net.neoforged.neoforge.transfer.resources.ItemResource resource, int amount){
```

Comment:
Minor nitpick, missing space before brace at end of line, and missing space before `new ArrayList` on the following line.

Also as vanilla doesn't have similar with methods for updating a single slot, we should evaluate if we want to do this, or if we would rather have this be a private helper in the singular calling class: `ItemContainerContentsResourceHandler` (which would also save us having to patch this method in)